modules/translate.py

```diff
@@ -0,0 +1,176 @@
+import os
+from math import floor
+from sys import getsizeof
+
+from google.cloud import translate
+from langchain_core.documents import Document
+from langchain_google_community import GoogleTranslateTransformer
+
+
+def truncate_text(text, limit=100000):
+    """
+    A simple function to truncate input text meet length requirements of Google translation's language detection API.
+
+    Args:
+        text (str): Text to be truncated.
+        limit (int): Maximum number of bytes allowed.
+    """
+    factor = limit / getsizeof(text.encode("utf-8"))
+    truncated = text[:floor(len(text) * factor)]
+    return truncated
+
+
+def detect_language(
+        text: str,
+        project_id: str = "llms-as-experts",
+        location: str = "global",
+):
+    """
+    Detect text language.
+    Args:
+        text (str): Text to be detected.
+        project_id (str): The ID of the project that owns the destination bucket.
+        location (str): Translation service location. Default to "global".
+    """
+
+    client = translate.TranslationServiceClient()
+    parent = f"projects/{project_id}/locations/{location}"
+    content = truncate_text(text)
+    response = client.detect_language(
+        content=content,
+        parent=parent,
+        mime_type="text/plain",
+    )
+    source_language_code = response.languages[0].language_code
```

Comment:
this seems like a good spot for some error handling, if you found it very robust in returning a language code feel free to ignore